src/pigpio.cc

```diff
@@ -687,6 +707,205 @@ NAN_METHOD(gpioNotifyClose) {
   }
 }
 
+/* ------------------------------------------------------------------------ */
+/* Waves                                                                    */
+/* ------------------------------------------------------------------------ */
+
+NAN_METHOD(gpioWaveClear) {
+  int rc = gpioWaveClear();
+  if (rc < 0) {
+    return ThrowPigpioError(rc, "gpioWaveClear");
+  }
+  info.GetReturnValue().Set(rc);
+}
+
+NAN_METHOD(gpioWaveAddNew) {
+  int rc = gpioWaveAddNew();
+  if (rc < 0) {
+    return ThrowPigpioError(rc, "gpioWaveAddNew");
+  }
+  info.GetReturnValue().Set(rc);
+}
+
+NAN_METHOD(gpioWaveAddGeneric) {
```

Comment:
It should not be necessary to explicitly pass the number of pulses to `gpioWaveAddGeneric` as the first argument. This can be implicitly determined by the array size.

Reply:
Fixed C source ,JS source and Readme wiith 54357ae